src/hub/Hub.sol

```diff
@@ -615,16 +627,33 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler, IHubGuar
     }
 
     /// @inheritdoc IHubGatewayHandler
-    function increaseShareIssuance(PoolId poolId, ShareClassId scId, uint128 amount) external {
+    function initiateTransferShares(
+        uint16 centrifugeId,
+        PoolId poolId,
+        ShareClassId scId,
+        bytes32 receiver,
+        uint128 amount
+    ) external {
+        _auth();
+
+        require(hubRegistry.chain(poolId, centrifugeId), DisabledChain());
+        emit ForwardTransferShares(centrifugeId, poolId, scId, receiver, amount);
+        sender.sendExecuteTransferShares(poolId, scId, centrifugeId, receiver, amount);
+    }
+
+    /// @inheritdoc IHubGatewayHandler
+    function increaseShareIssuance(uint16 centrifugeId, PoolId poolId, ShareClassId scId, uint128 amount) external {
         _auth();
 
+        require(hubRegistry.chain(poolId, centrifugeId), DisabledChain());
         shareClassManager.increaseShareClassIssuance(poolId, scId, amount);
     }
 
     /// @inheritdoc IHubGatewayHandler
-    function decreaseShareIssuance(PoolId poolId, ShareClassId scId, uint128 amount) external {
+    function decreaseShareIssuance(uint16 centrifugeId, PoolId poolId, ShareClassId scId, uint128 amount) external {
         _auth();
 
+        require(hubRegistry.chain(poolId, centrifugeId), DisabledChain());
         shareClassManager.decreaseShareClassIssuance(poolId, scId, amount);
     }
```

Comment:
I think we have the same issue with these messages. This message does not act over an assetId. We're checking here the poolId matches with the centrifugeId, but this messages is Vault to Hub, so any bad adapter can sent this message and pass the check.

Reply:
I think this issue will disappear once we configure adapters by pool ID. Actually do we really need this PR, if we go with adapters-per-pool?